flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/table/config/BigQueryReadTableConfig.java

```diff
@@ -143,7 +162,7 @@ public BigQueryReadTableConfig.Builder credentialFile(String credentialFile) {
 
         @Override
         public BigQueryReadTableConfig.Builder testMode(Boolean testMode) {
-            super.testMode = this.testMode;
+            super.testMode = testMode;
```

Comment:
Seems like a bug in previous code. How did it behave before your change?

Reply:
Yes, took the default value always. 
Tests are important. :)